SplitFiles/game.c

```diff
@@ -44,8 +44,8 @@ int wait_for_a_key(GameComponent* input, GameParameters* params) {
 void tick_func(GameComponent* input, GameParameters* params) { //Now a scoreboard too
   int last_score = (int)input->ptr;
   if (params->score != last_score) {
-    screenNumber(25, 1, params->score);
-    input->ptr = (void *)params->score;
+    screenNumber(23, 1, params->score);
+    input->ptr = (void *)last_score;
```

Comment:
eeek - I don't know what this (line 48) is!

Reply:
It's a hack, storing the last score in the pointer rather than creating a struct with one int in it. The version of line 48 you just overwrite (ie in red) is the correct one. The one in green has a bug meaning the score is updated on screen every loop